kolibri/core/assets/src/utils/i18n.js

```diff
@@ -0,0 +1,212 @@
+  }
+  $formatTime(time, options = {}) {
+    return vue.prototype.$formatTime(time, options);
+  }
+  $formatRelative(date, options = {}) {
+    return vue.prototype.$formatRelative(date, options);
+  }
+  $formatNumber(number, options = {}) {
+    return vue.prototype.$formatNumber(number, options);
+  }
+  $formatPlural(plural, options = {}) {
+    return vue.prototype.$formatPlural(plural, options);
+  }
+}
+
+/**
+ * Returns a Translator instance.
+ * @param {string} nameSpace - The nameSpace of the messages for translation.
+ * @param {object} defaultMessages - an object mapping message ids to default messages.
+ */
+export function createTranslator(nameSpace, defaultMessages) {
+  return new Translator(nameSpace, defaultMessages);
+}
+
+function setUpVueIntl() {
+  /**
+   * Use the vue-intl plugin.
+   **/
+  vue.use(VueIntl, { defaultLocale });
+
+  vue.prototype.isRtl = global.languageDir === 'rtl';
+
+  languageDirection = global.languageDir || languageDirection;
+
+  if (global.languages) {
+    Object.assign(availableLanguages, global.languages);
+  }
+
+  vue.prototype.$tr = function $tr(messageId, args) {
+    const nameSpace = this.$options.name || this.$options.$trNameSpace;
+    return $trWrapper(nameSpace, this.$options.$trs, this.$formatMessage, messageId, args);
+  };
+  vue.prototype.$trHtml = function $trHtml(messageId, args) {
+    const nameSpace = this.$options.name || this.$options.$trNameSpace;
+    return $trWrapper(nameSpace, this.$options.$trs, this.$formatHTMLMessage, messageId, args);
+  };
+
+  if (global.languageCode) {
+    currentLanguage = global.languageCode;
+    vue.setLocale(currentLanguage);
+    setLanguageDensity(currentLanguage);
+
+    if (global.coreLanguageMessages) {
+      vue.registerMessages(currentLanguage, global.coreLanguageMessages);
+    }
+  }
+}
+
+export function setUpIntl() {
+  /**
+   * If the browser doesn't support the Intl polyfill, we retrieve that and
+   * the modules need to wait until that happens.
+   **/
+  return new Promise((resolve, reject) => {
+    if (!Object.prototype.hasOwnProperty.call(global, 'Intl')) {
+      Promise.all([
+        new Promise(resolve => {
+          require.ensure([], require => {
+            resolve(() => require('intl'));
+          });
+        }),
+        importIntlLocale(global.languageCode),
+      ]).then(
+        // eslint-disable-line
+        requires => {
+          // Executes function that requires 'intl'
+          requires[0]();
+          // Executes function that requires intl locale data - needs intl to have run
+          requires[1]();
+          setUpVueIntl();
+          resolve();
+        },
+        error => {
+          logging.error(error);
+          logging.error('An error occurred trying to setup Internationalization', error);
+          reject();
+        }
+      );
+    } else if (global.languageCode === 'rt-lft') {
+      require.ensure([], () => {
+        toFakeRTL = require('./mirrorText').toFakeRTL;
```

Comment:
Yeah, not sure this is a huge issue as this is not intended to be production code.